### PR TITLE
Enable and fix Lint issues

### PR DESCRIPTION
**Pull request by bjk-soundcloud**
Enable and fix Lint issues:
* Lint/AmbiguousRegexpLiteral
* Lint/AssignmentInCondition
* Lint/ConditionPosition
* Lint/HandleExceptions
* Lint/UnusedBlockArgument
* Lint/UselessAssignment

### CODE REVIEW

**Comment by bjk-soundcloud** [comment]
@arthurnn More cleanups :)

File: lib/lhm/printer.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def write(message)
     end
 
     class Dot < Base
-      def notify(lowest = nil, highest = nil)
+      def notify(_lowest = nil, _highest = nil)
```
Comment (arthurnn) [inline lib/lhm/printer.rb]: why this change?

File: spec/integration/integration_helper.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@
 begin
   require 'active_support'
 rescue LoadError
+  print 'No "active_support" loaded'
```
Comment (arthurnn) [inline spec/integration/integration_helper.rb]: any clue why do we rescue this error at all? does things still work without Active support?

File: lib/lhm/printer.rb (hunk 1 above)
Comment (bjk-soundcloud) [inline lib/lhm/printer.rb]: looking at this block, it doesn't look like lowest and highest are used.  But possibly necessary because the extend the class, which is why I assume they were set to nil in the first place.

File: spec/integration/integration_helper.rb (hunk 2 above)
Comment (bjk-soundcloud) [inline spec/integration/integration_helper.rb]: No idea, if we only work with active support let's drop the begin/rescue block.

Comment (grobie) [inline spec/integration/integration_helper.rb]: This was here to run the integration tests for both datamapper as well as activerecord/activesupport. If support for datamapper has been dropped, the begin/rescue block as well as the `defined?` check cah be removed.

Comment (arthurnn) [inline spec/integration/integration_helper.rb]: cool.. lets not do this right now, as we didnt drop support yet.. but thanks @grobie for the info.

File: lib/lhm/printer.rb (hunk 1 above)
Comment (arthurnn) [inline lib/lhm/printer.rb]: how about we change this to `notify(*)` ?

Comment (bjk-soundcloud) [inline lib/lhm/printer.rb]: Works for me.

File: spec/integration/integration_helper.rb (hunk 2 above)
Comment (arthurnn) [inline spec/integration/integration_helper.rb]: Support for DM has been dropped on master, can you remove this rescue and rebase the branch?

thanks

**Comment by bjk-soundcloud** [comment]
@arthurnn merged, updated, and poked the tests.

**Comment by arthurnn** [comment]
thanks :heart:
